dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/persistence/file/datastore/DataStoreFileReaderTest.kt

```diff
@@ -0,0 +1,289 @@
+@ForgeConfiguration(Configurator::class)
+internal class DataStoreFileReaderTest {
+    private lateinit var testedDatastoreFileReader: DatastoreFileReader
+
+    @Mock
+    lateinit var mockDataStoreFileHelper: DataStoreFileHelper
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Mock
+    lateinit var mockTLVBlockFileReader: TLVBlockFileReader
+
+    @Mock
+    lateinit var mockDataStoreFile: File
+
+    @TempDir
+    lateinit var mockStorageDir: File
+
+    @Mock
+    lateinit var mockDataStoreDirectory: File
+
+    @Mock
+    lateinit var mockDeserializer: Deserializer<String, ByteArray>
+
+    @StringForgery
+    lateinit var fakeFeatureName: String
+
+    @StringForgery
+    lateinit var fakeDataString: String
+
+    @StringForgery
+    lateinit var fakeKey: String
+
+    private lateinit var fakeDataBytes: ByteArray
+    private lateinit var versionBlock: TLVBlock
+    private lateinit var dataBlock: TLVBlock
+    private lateinit var blocksReturned: ArrayList<TLVBlock>
+
+    @BeforeEach
+    fun setup() {
+        fakeDataBytes = fakeDataString.toByteArray(Charsets.UTF_8)
+
+        whenever(
+            mockDataStoreFileHelper.getDataStoreFile(
+                featureName = eq(fakeFeatureName),
+                storageDir = eq(mockStorageDir),
+                key = any()
+            )
+        ).thenReturn(mockDataStoreFile)
+
+        whenever(mockDataStoreDirectory.existsSafe(mockInternalLogger)).thenReturn(true)
+        whenever(mockDataStoreFile.existsSafe(mockInternalLogger)).thenReturn(true)
+        whenever(mockDeserializer.deserialize(fakeDataString)).thenReturn(fakeDataBytes)
+
+        versionBlock = createVersionBlock(true)
+        dataBlock = createDataBlock()
+        blocksReturned = arrayListOf(versionBlock, dataBlock)
+        whenever(mockTLVBlockFileReader.read(mockDataStoreFile)).thenReturn(blocksReturned)
+
+        testedDatastoreFileReader = DatastoreFileReader(
+            dataStoreFileHelper = mockDataStoreFileHelper,
+            featureName = fakeFeatureName,
+            internalLogger = mockInternalLogger,
+            storageDir = mockStorageDir,
+            tlvBlockFileReader = mockTLVBlockFileReader
+        )
+    }
+
+    @Test
+    fun `M return no data W read() { datastore file does not exist }`() {
+        // Given
+        whenever(mockDataStoreFile.existsSafe(mockInternalLogger)).thenReturn(false)
+
+        // When
+        testedDatastoreFileReader.read(
+            key = fakeKey,
+            deserializer = mockDeserializer,
+            callback = object : DataStoreCallback<ByteArray> {
+                override fun onSuccess(dataStoreContent: DataStoreContent<ByteArray>?) {
+                    assertThat(dataStoreContent).isNull()
+                }
+
+                override fun onFailure() {
+                    // should not get here
+                    assertThat(1).isEqualTo(2)
+                }
+            }
+        )
```

Comment:
there is a better way to deal with such assertion (`assertThat(1).isEqualTo(2)` looks very strange, tbh). You need to use a mock and `argumentCaptor`

```suggestion
        // Given
        whenever(mockDataStoreFile.existsSafe(mockInternalLogger)).thenReturn(false)
        val mockCallback = mock<DataStoreCallback<ByteArray>>()

        // When
        testedDatastoreFileReader.read(
            key = fakeKey,
            deserializer = mockDeserializer,
            callback = mockCallback
        )

        // Then
        argumentCaptor<DataStoreContent<ByteArray>?> {
          verify(mockCallback).onSuccess(capture())
          assertThat(lastValue).isNull
          verifyNoMoreInteractions(mockCallback)
        }
```

Same approach can be applied to the tests below as well.